src/components/Calculator.js

```diff
@@ -1,34 +1,44 @@
 import React from 'react';
+import '../css/Calculator.css';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
     this.state = {};
+    this.onClickHandler = this.onClickHandler.bind(this);
+  }
+
+  onClickHandler(e) {
+    this.setState((state) => calculate(state, e.target.innerText));
   }
 
   render() {
+    const { next, total } = this.state;
     return (
       <div className="wrapper">
-        <div className="total">0</div>
-        <div className="ac">AC</div>
-        <div className="plus-minus">+/-</div>
-        <div className="percent">%</div>
-        <div className="divide">/</div>
-        <div className="seven">7</div>
-        <div className="eight">8</div>
-        <div className="nine">9</div>
-        <div className="multiply">*</div>
-        <div className="four">4</div>
-        <div className="five">5</div>
-        <div className="six">6</div>
-        <div className="minus">-</div>
-        <div className="one">1</div>
-        <div className="two">2</div>
-        <div className="three">3</div>
-        <div className="plus">+</div>
-        <div className="zero">0</div>
-        <div className="point">.</div>
-        <div className="equal">=</div>
+        <div className="total">
+          {next || total || 0}
```

Comment:
Currently, the user can't see what operation they have chosen. 


https://user-images.githubusercontent.com/34578631/171605678-93663e8e-daf0-47bc-b549-80acd21c9a3b.mov


Kindly, add `operation` in line 17 and use something like this instead of line 21 👇 

```
{ total }
{ operation }
{ next }

``` 
This way will display all the necessary inputs for the user and don't forget to set the initial value of `0` for the `total` in line 8 that way it will start always begin at zero and try to figure to set zero when the `AC` button is clicked. 
**Hint:** change the state inside the `calculate.js` file for the `Ac` button.

Reply:
Thank you for detailed feedback!